llarp/ev/ev_libuv.cpp

```diff
@@ -1026,55 +530,27 @@ namespace libuv
   }
 
   bool
-  Loop::udp_close(llarp_udp_io* udp)
+  Loop::add_network_interface(
+      std::shared_ptr<llarp::vpn::NetworkInterface> netif,
+      std::function<void(llarp::net::IPPacket)> handler)
   {
-    if (udp == nullptr)
-      return false;
-    auto* glue = static_cast<udp_glue*>(udp->impl);
-    if (glue == nullptr)
-      return false;
-    glue->Close();
-    return true;
-  }
-
-  bool
-  Loop::tun_listen(llarp_tun_io* tun)
-  {
-#if defined(_WIN32) || defined(_WIN64)
-    (void)tun;
-    return false;
-#else
-    auto* glue = new tun_glue(tun);
-    tun->impl = glue;
+    auto* glue = new tun_glue(netif, handler);
```

Comment:
implicit pointer ownership transfer is nasty.

Reply:
if not feasible to make the ownership transfer more explicit, leave a comment here saying why we're *apparently* new-ing a pointer and dropping it on the floor.